library/core/validation/src/main/kotlin/de/lise/fluxflow/validation/jakarta/JakartaDataValidationBuilder.kt

```diff
@@ -73,7 +75,15 @@ class JakartaDataValidationBuilder(
         }
 
         prop.findAnnotationEverywhere<Valid>() ?: return result
-        val propertyType = prop.returnType.classifier as? KClass<*> ?: return result
+
+        val propTypeClassifier = prop.returnType.classifier as KClass<*>? ?: return result
+
+        val propertyType: KClass<*> = if (isCollectionType(propTypeClassifier)) {
+            // We need to get the type of the collection elements and not the collection itself
+            prop.returnType.arguments.first().type!!.classifier as? KClass<*> ?: return result
```

Comment:
:+1: That is definitely more correct than the previous version. But maybe we need both:
1. the validation constraints on the generic type itself **and**
2. the validation constraints found on the generic type arguments (of which there could be more than one - eg. `Map<TKey, TValue>`

I'd suggest creating another issue, that extends your PR.

Reply:
https://github.com/lisegmbh/fluxflow/issues/184